plugins/datadog/datadog.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	log "github.com/Sirupsen/logrus"
+	plugin "github.com/dcos/dcos-metrics/plugins"
+	"github.com/dcos/dcos-metrics/producers"
+	"github.com/urfave/cli"
+)
+
+func handleErr(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	log.Info("Starting datadog DC/OS metrics plugin")
+
+	ddPluginFlags := []cli.Flag{
+		cli.StringFlag{
+			Name:  "datadog-host",
+			Value: "localhost",
+			Usage: "Datadog URL to query",
+		},
+		cli.IntFlag{
+			Name:  "datadog-port",
+			Value: 8125,
+			Usage: "Datadog port",
+		},
+	}
+
+	ddp, err := plugin.New(ddPluginFlags)
+	handleErr(err)
```

Comment:
it seems that you call this function only once, maybe we should  get rid of it and make
`log.Fatalf("Unable to initialize a new plugin: %s", err)`